boa/vm/fork.py

```diff
@@ -37,29 +37,38 @@ class CachingRPC(RPC):
     def __init__(self, rpc: RPC, cache_file: str = DEFAULT_CACHE_DIR):
         # (default to memory db plyvel not found or cache_file is None)
         self._rpc = rpc
-        self._init_mem_db()
-        if cache_file is not None:
-            try:
-                from boa.util.leveldb import LevelDB
-
-                print("(using leveldb)", file=sys.stderr)
-
-                cache_file = os.path.expanduser(cache_file)
-                # use CacheDB as an additional layer over disk
-                # (ideally would use leveldb lru cache but it's not configurable
-                # via LevelDB API).
-                self._db = CacheDB(LevelDB(cache_file), cache_size=1024 * 1024)  # type: ignore
-            except ImportError:
-                # plyvel not found
-                pass
+        #Only attempt DB initialization if its not done prior
+        #It appears __init__ is being called multiple times, which causes LOCK file error from levelDB.
```

Comment:
raising an exception if the LOCK file is there is the correct behavior, because leveldb is not concurrency safe.